sdk/core/core-http/src/policies/systemErrorRetryPolicy.ts

```diff
@@ -172,17 +172,19 @@ function retry(
       err.code === "ENOENT")
   ) {
     // If previous operation ended with an error and the policy allows a retry, do that
-    return utils
-      .delay(retryData.retryInterval)
-      .then(() => policy._nextPolicy.sendRequest(request.clone()))
-      .then((res) => retry(policy, request, res, retryData, err))
-      .catch((err) => retry(policy, request, operationResponse, retryData, err));
+    try {
+      await utils.delay(retryData.retryInterval);
+      return policy._nextPolicy.sendRequest(request.clone());
+    }
+    catch (err) {
+      return retry(policy, request, operationResponse, retryData, err);
+    }
   } else {
     if (err != undefined) {
```

Comment:
nit: I think this should just be `if (err)` to avoid strict equals lint problems

Reply:
Doned